src/pubsub/ua_pubsub_ns0.c

```diff
@@ -826,6 +826,15 @@ addWriterGroupRepresentation(UA_Server *server, UA_WriterGroup *writerGroup){
     UA_Server_writeAccessLevel(server, contentMaskId,
                                UA_ACCESSLEVELMASK_WRITE | UA_ACCESSLEVELMASK_READ);
 
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS
+    retVal |= UA_Server_addReference(server, writerGroup->identifier,
+                                     UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                                     UA_EXPANDEDNODEID_NUMERIC(0, UA_NS0ID_WRITERGROUPTYPE_ADDDATASETWRITER), true);
+    retVal |= UA_Server_addReference(server, writerGroup->identifier,
+                                     UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                                     UA_EXPANDEDNODEID_NUMERIC(0, UA_NS0ID_WRITERGROUPTYPE_REMOVEDATASETWRITER), true);
+#endif
+
```

Comment:
The open62541 targets also resource constraint devices. To reduce the memory consumption the default cmake settings are loading a customized reduced nodeset. This reduced nodeset is also available for PubSub (see tools/schema). The removeDataSetWriter method is currently not contained in this minimal nodeset. It would be nice if you can add the missing parts (copy all needed parts from the Nodeset2.xml) and test your changes with the reduced and full nodeset.

Reply:
I moved the reference to removeDataSetWriter under WriterGroupType out of the comment and added the RemoveDataSetWriter Node with its input argument. I tested with `set(UA_NAMESPACE_ZERO "REDUCED")` in my project and i was still able to add and delete a WriterGroup. Let me know if i missed anything/did something wrong.